lib/g_searcher/search_results/report_search_result.ex

```diff
@@ -0,0 +1,21 @@
+defmodule GSearcher.SearchResults.ReportSearchResult do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias GSearcher.SearchResults.{Report, SearchResult}
+
+  schema "reports_search_results" do
+    belongs_to :report, Report
+    belongs_to :search_result, SearchResult
```

Comment:
I'm not sure we really need this table as we could move the `report_id` into the `search result table` 🤔 , what is the point to have this table @bterone 🤔, a `search result` should belong to a `report` and a report might have many `search results`, 2 tables are enough for me.

Reply:
This was because I intentionally wanted a `many-to-many` relationship. This way we could reuse `search_results` if multiple users happen to want to search the same `keyword`, or as you said before, a user just wanted to update their CSV file a little bit and reupload it. 😂 

Expanding this idea further, we could check if the record was `old enough` where it warrants a new query to Google. 
It's just a way to help prevent querying Google too much.

Reply:
I see,

IMO I think we shouldn't `cache` the Google search result like that, the search result should become from Google at the time the user uploads that, as the data is always changing on Google 😀  so using the old `search` data isn't a good feature.

Another thing, I would like to clarify something

1/ `This way we could reuse search_results if multiple users happen to want to search the same keyword` -> Not sure about this as the `report` belongs to a User so how could we map this user to another user's search keyword 🤔

2/ `as you said before, a user just wanted to update their CSV file a little bit and reupload it.` -> we should give them the latest search result in this case, for example, they re-upload it after 1-week :D.

3/ `we could check if the record was old enough where it warrants a new query to Google.` -> Well, do we have the `old enough` number yet? 1 week, 1 month it hards right? 😀 so I prefer to just go as a simple feature first 😀

4/ Also for the `we could check if the record was old enough where it warrants a new query to Google.` we could achieve with the `search_result` table by the `created_at` record :D 

Am I missing something? 🤔

Reply:
To answer your questions,

1/ Since we separated `reports` and `search_results`, the `reports_search_results` just contains the IDs of both and we can associate by a `many-to-many` relationship.

2 + 3/ I haven't really researched the topic much, I only quickly googled it before on how quickly google index sites and it was `between four days and four weeks` so I was planning for `1 day` as `old enough`, however looking further, they have ways to crawl for updates in a matter of hours or even [minutes](https://googleblog.blogspot.com/2011/11/giving-you-fresher-more-recent-search.html) for something like news articles.
They use different algorithms depending on what is the type of content
```
The weight applied to each factor varies depending on the nature of your query—for example, 
the freshness of the content plays a bigger role in answering queries 
about current news topics than it does about dictionary definitions.
```
https://www.google.com/search/howsearchworks/algorithms/

4/ Yeah that was the exact idea I wanted to go with 👍 

In regards to how the data was associated, the main question I was asking myself was;
A **report** can have `many` **search_results**, but can a **search_result** belong to many **reports**?

I agree with simple is usually better but I was thinking in terms of only if its **possible** to reuse search results in multiple reports, since DB changes require extra migrations, schema changes, and scripts to migrate the data which is more work in the long run.

I think since the content can be minutes old depending on what we're searching for, I don't mind creating an improvement to just make it `one-to-many` for the report/search_result 🤔

Reply:
Haha sure, if you have some idea for the long run already then I'm fine with that @bterone :)